### ISSUE TITLE
Prevent ActiveRecord::Deadlocked With Circular Article/Collection Updates

### ISSUE THREAD

**Issue by mstruve** [thepracticaldev/dev.to#5964]
When an Article is updated if it belongs to a collection we will then `touch` that collection object. 
```ruby
class Article < ApplicationRecord
  belongs_to :collection, optional: true, touch: true
end
```
When a collection is touched, it then touches all of its related articles including the article that was just updated
```ruby
class Collection < ApplicationRecord
  after_touch :touch_articles
end
```
This sometimes leads to a deadlock when we go to update an article. 
Honeybadger: https://app.honeybadger.io/fault/66984/0d294eb872c932b638e97fed31595826
```
ActiveRecord::Deadlocked: PG::TRDeadlockDetected
```

We need to reorganize these callbacks in such a way that we don't update an article and then try to update it again after updating the respective collection.

**Comment by gonzar11** [comment]
Hey @mstruve can we remove `touch: true` option from association and just touch the collection inside of `after_commit` callback. .Something like this:

**Comment by mstruve** [comment]
I am assuming it is there for a reason but I am not sure as I have not really dug into that code. If you think that is an option then definitely explore it and dig in more to see if its needed.

**Comment by gonzar11** [comment]
HI @mstruve sorry for the delay. I've implemented a possible solution.

**Comment by bolariin** [comment]
I think this might have been resolved by #6371

**Comment by rhymes** [comment]
Thanks a lot for the find @bolariin!!
